File: frontend/src/types/schoolEvent.ts
export type EventCategory = 'EXAM' | 'PROJECT' | 'TUITION' | 'EVENT' | 'DEADLINE';

export interface SchoolEvent {
  id: string;
  title: string;
  date: string;
  category: EventCategory;
  estimatedCost: number | null;
  notes: string | null;
  daysUntil: number;
  isUpcoming: boolean;
  budgetSuggestion: string | null;
  createdAt: string;
}

export interface SchoolEventPayload {
  title: string;
  date: string;
  category: EventCategory;
  estimatedCost?: number | null;
  notes?: string;
}

File: src/main/java/com/iponchallenge/entity/SchoolEvent.java
package com.iponchallenge.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "school_events")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SchoolEvent {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private LocalDate date;

    // EXAM | PROJECT | TUITION | EVENT | DEADLINE
    @Column(nullable = false)
    private String category;

    // Optional expected cost for budget suggestion
    @Column(name = "estimated_cost", precision = 12, scale = 2)
    private BigDecimal estimatedCost;

    @Column
    private String notes;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
